File: swig/csharp/tests/test10.cs
/*
 * Copyright 2006-2012 The FLWOR Foundation.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
using System;
using System.Text;
using org.zorbaxquery.api;

namespace ZorbaApplication
{

    class MyDiagnosticHandler : DiagnosticHandler {
        public override void error(ZorbaException exception) {
          System.Console.WriteLine("Exception catched in MyDiagnosticHandler");
          System.Console.WriteLine("Description: " + exception.getDescription());
        }
    }

    class Program
    {
        static String test() {
            MyDiagnosticHandler diagnosticHandler = new MyDiagnosticHandler();

            InMemoryStore store = InMemoryStore.getInstance();
            Zorba zorba = Zorba.getInstance(store);

            ItemFactory itemFactory = zorba.getItemFactory();
            Item iNull = itemFactory.createJSONNull();
            System.Console.WriteLine( "Creating JSON Item null:  " +  iNull.serialize());
            iNull.Dispose();
            
            Item iNumber = itemFactory.createJSONNumber("5");
            System.Console.WriteLine( "Creating JSON Item number:  " + iNumber.serialize() );
            iNumber.Dispose();
            
            StringVector sv = new StringVector();
            sv.Add("Hello");
            sv.Add("Zorba");
            sv.Add("Hello");
            sv.Add("Zorba");
            Item iArray = itemFactory.createJSONArray(sv);
            sv.Dispose();
            System.Console.WriteLine( "Creating JSON Item array:  " + iArray.serialize() );
            iArray.Dispose();
            
            zorba.shutdown();
            InMemoryStore.shutdown(store);
            
            return "ok";
        }

        static void Main(string[] args)
        {
            System.Console.WriteLine("Running: Compile query string using Diagnostic Handler");

            String testResult = "ok";
            String result;
            try {
                result = test();
            } catch(Exception e) {
                System.Console.WriteLine("Failed");
                Console.WriteLine("{0} Exception caught.", e);
                return;
            }
            
            if (result.Equals(testResult)) {
                System.Console.WriteLine("Success");
            } else {
                System.Console.WriteLine("Failed");
            }

            
        }
    }
}

File: swig/php/tests/test10.php
<?
/*
 * Copyright 2006-2013 The FLWOR Foundation.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * 
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

require 'zorba_api_wrapper.php';

class MyDiagnosticHandler extends DiagnosticHandler {
  public function error($args) {
    print "Error caught in DiagnosticHandler: " . $args->getDescription() . "\n";
  }
}

function test(Zorba $aZorba, $query)
{
  $diagnosticHandler = new MyDiagnosticHandler();
  $xquery = $aZorba->compileQuery($query, $diagnosticHandler);
  try {
    print $xquery->execute();
  } catch (Exception $e) {
    print "Caught error: " . $e;
    exit();
  }
  $xquery->destroy();
  return;
}

$store = InMemoryStore::getInstance();
$zorba = Zorba::getInstance($store);

print "Running: Compile query string using JSONiq \n";

$query =  "";
$query .= " let $sats:=jn:json-doc(\"@phpPath@/tests/satellites.json\")('satellites')  ";
$query .= "  return { ";
$query .= "    'visible' : [  ";
$query .= "       for $sat in jn:keys($sats)  ";
$query .= "       where $sats($sat)('visible')  ";
$query .= "       return $sat  ";
$query .= "    ],  ";
$query .= "    'invisible' : [  ";
$query .= "       for $sat in jn:keys($sats)  ";
$query .= "       where not($sats($sat)('visible'))  ";
$query .= "       return $sat  ";
$query .= "    ]  ";
$query .= "  } ";

test($zorba, $query);
print "Success\n";

$zorba->shutdown();
InMemoryStore::shutdown($store);

?>
